Orange/widgets/data/owsave.py

```diff
@@ -22,58 +38,126 @@ class Inputs:
     resizing_enabled = False
 
     last_dir = Setting("")
-    last_filter = Setting("")
     auto_save = Setting(False)
+    filetype = Setting(FILE_TYPES[0][0])
+    compression = Setting(COMPRESSIONS[0][0])
+    compress = Setting(False)
+
+    def get_writer_selected(self):
+        writer = FileFormat.get_reader(self.type_ext)
+        try:
+            writer.EXTENSIONS = [
+                writer.EXTENSIONS[writer.EXTENSIONS.index(self.type_ext + self.compress_ext)]]
```

Comment:
You still have the problem mentioned in https://github.com/biolab/orange3/pull/3147#discussion_r206903890

Instead of the try/except, you should just check
```py
ext = self.type_ext + self.compress_ext
if ext not in writer.EXTENSIONS:
    self.Error.not_supported_extension()
    return None
writer.EXTENSIONS = [ext]
return writer
```

BTW, just a minor suggestion, but a better name for `not_supported_extension` would be `unsupported_extension`.